classes/class-schedule.php

```diff
@@ -965,6 +967,79 @@ protected function do_action( $action ) {
 	}
 
 	/**
+	 * Calculate schedule run time.
+	 *
+	 * @param int Timestamp $end
+	 */
+	protected function update_average_schedule_run_time( $end ) {
+
+		$start = $this->get_schedule_running_start_time();
+
+		if ( $end <= $start ) {
+			// Something went wrong, ignore.
+			return;
+		}
+
+		$diff = (int) abs( $end - $start );
+
+		if ( isset( $this->options['duration_total'] ) && isset( $this->options['backup_run_count'] ) ) {
+
+			$this->options['duration_total'] += $diff;
+			$this->options['backup_run_count'] ++;
+
+		} else {
+
+			$this->options['duration_total'] = $diff;
+			$this->options['backup_run_count'] = 1;
+
+		}
+
+		$this->save();
+	}
+
+	/**
+	 * Calculates the average run time for this schedule.
+	 *
+	 * @return string
+	 */
+	public function get_schedule_average_duration() {
+
+		if ( ! isset( $this->options['backup_run_count'] ) || ! isset( $this->options['backup_run_count'] )  ) {
+			return '0';
+		}
+
+		if ( 0 === (int) $this->options['backup_run_count'] ) {
+			return '0';
```

Comment:
Would be good to show something other than `0` if the backup hasn't run yet as `0` makes it seem like the average run time is `0` which is probably misleading. Instead I'd show something like `N/A`.